chall04/fulldemo.py

```diff
@@ -0,0 +1,80 @@
+#! /usr/local/bin/python3
+
+import sys
+import fileinput
+
+def error(e, n):
+	if e == 0:
+		if n != 0:
+			print("%s: %s: Can't read file" % (sys.argv[0], sys.argv[n]))
+		else:
+			print("%s: stdin: Can't read file" % (sys.argv[0]))
+	else:
+		if n != 0:
+			print("%s: %s: Not enough space in the given shelves" % (sys.argv[0], sys.argv[n]))
+		else:
+			print("%s: stdin: Not enough space in the given shelves" % (sys.argv[0]))
+
+def shelves_to_use(all_file):
+	shelves_use = 0
+	for i in range(1, len(all_file)):
+		line = all_file[i].split()
+		if len(line) < 2:
+			return (-1)
+		try:
+			shelves_use = int(line[0]) + shelves_use
+			if int(line[0]) < 0:
+				return (-1)
+		except:
+			return (-1)
+	return(shelves_use)
```

Comment:
Why are you merging the shelves? You won't be able to differentiate them later :/